ml-agents/mlagents/trainers/agent_processor.py

```diff
@@ -16,31 +17,39 @@ class AgentProcessor:
     One AgentProcessor should be created per agent group.
     """
 
-    def __init__(self, trainer: Trainer, policy: TFPolicy, max_trajectory_length: int):
+    def __init__(
+        self,
+        trainer: Trainer,
+        policy: TFPolicy,
+        max_trajectory_length: int,
+        stats_category: str,
```

Comment:
If this is coming from the trainer, why not pass a `StatsReporter`?

Reply:
Done